backend/src/books/books.service.ts

```diff
@@ -1,13 +1,60 @@
 import { Injectable, HttpService } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import * as Papa from 'papaparse';
+import { Readable } from 'stream';
+import { validate, ValidationError } from 'class-validator';
 
 import { BookQueryResponse } from './books';
-import { Book } from '../../../interfaces/books';
+import { Book as BookType } from '../../../interfaces/books';
+import { Book } from './book.entity';
+import { BookDto } from './book.dto';
+
+/** https://stackoverflow.com/a/54136803 */
+const bufferToStream = (buffer: Buffer): Readable => {
+  const readableStream = new Readable({
+    read(): void {
+      this.push(buffer);
+      this.push(null);
+    },
+  });
+
+  return readableStream;
+};
+
+const validateBooks = async (
```

Comment:
Nestjs does automatically the validation if you provde the `Dto` definition to the controller.

Reply:
Yes, but it doesn't work in this case because we are sending a file in form-data and not JSON in the body

Reply:
Fair enough!